add support for reading typst pagebreak

RayBB opened the pull request:
Closes #11101

jgm commented:
If we are using a div here, we might consider adding `wrapper="1"`, so that it can be "unwrapped" in formats that allow putting an attribute directly on a horizontal rule.

@tarleb does that make sense?

For the class name, should it be `pagebreak` or `page-break`?

Or should we go whole hog and add PageBreak to the AST?

tarleb commented:
> Or should we go whole hog and add PageBreak to the AST?

I'll probably have about a week of quiet time in which I should be able to work on some of those bigger changes. I guess we'd want to bundle a couple of AST changes into a single release?

RayBB commented:
It seems that you both are still considering a different approach of adding this to the AST more broadly.

I'm not sure if that means you want to merge this PR in the mean time or just wait.

In any case, I renamed pagebreak to page-break and added wrapper=1 so I think your suggestions are addressed if you do want to merge this. @tarleb

tarleb commented:
Excellent, thanks! I think the plan is to merge this, we're just thinking out loud about possible alternatives and future changes.
